### PR TITLE
v0.17.0 - (CPD-10886) userAuth module from GlobalWeb

### PR DESCRIPTION
**Pull request by xander-marjoram**
Movement of the userAuth module from globalweb.

## Platform Review Checks
- [ ] The code has been tested and reviewed on a QA Server
- [x] Unit tests have been written/updated and run (as required)

### CODE REVIEW

**Comment by ashleynolan** [comment]
@xander-marjoram Any way of ensuring test coverage stays at 100%?

**Comment by xander-marjoram** [comment]
@ashleynolan 💯 🔥 🚀 🌔

**Comment by ashleynolan** [comment]
@xander-marjoram 🎯 🏆 🥇

File: src/js/modules/userAuth/userData.js [hunk 1]
```diff
@@ -25,15 +25,6 @@ const fetchOrderCountAndSave = userData => {
             method: 'GET',
             credentials: 'same-origin'
         })
-            .then(response => {
-                if (response.ok) {
```
Comment (DamianMullins) [inline src/js/modules/userAuth/userData.js]: So, just so we record something around this, the issue with this block was that we couldn't get the unit tests to fall into this branch? Would be good to look into this a little more when work on the `f-fetch` kicks off as it looks like something which we'd likely need to be able to test against. 

Also, it looks like this change means that we are no longer passing the `statusText` along with the thrown error, was there a specific reason why were doing that in the first place?

**Comment by DamianMullins** [comment]
I forgot to mention but the changes you made to the header HTML (adding the order count and order count supported elements) should be added to [the templates in this module](https://github.com/justeat/f-header/blob/master/src/templates/header/index.hbs).

File: package.json [hunk 2]
```diff
@@ -28,6 +28,8 @@
     "@justeat/fozzie-colour-palette": "^0.9.0",
     "@kickoff/utils.scss": "^2.1.1",
     "include-media": "^1.4.9",
+    "jest-fetch-mock": "^1.4.1",
```
Comment (DamianMullins) [inline package.json]: I've just noticed that these modules should be in the `devDependencies` below..

**Pull request merged**